dsconvert.js

```diff
@@ -142,7 +142,8 @@ export function DesyncedStringToObject(str, info)
 					else
 					{
 						if (buf[p] == MP_DESYNCED_DEADKEY) { p++; GetIntPacked(); continue; } // used for Lua table memory layout
-						t[Parse(true)] = val;
```

Comment:
Bernhard implemented this with:

```
t[/\D/.test(key) ? key : ((key|0) - 1)] = val;
```

Reply:
I don't see how that can work. `/\D/.test(1)` returns false.

Reply:
The expression `/\D/.test(key) ? key : ((key|0) - 1` subtracts 1 from the key when `/\D/.test(1)` returns false, no?

Reply:
Oh right, that's a capital D. Still seems weird to be using a regex against what will be a Number object instead of a String, but I guess that does work.

Reply:
I agree, perhaps something to do with negative numbers? The regexp doesn't match negative numbers, whereas the other kind of checks would. But I don't know. It works, and matching what DesyncedJavaScriptUtils does seems desirable as long as it does work, and might help in the grand scheme of things if we find a bug.